internal/operator/namespace_test.go

```diff
@@ -12,46 +12,50 @@ import (
 )
 
 var _ = Describe("Namespace", func() {
+	var operatorClient operatorClient
 	logger.InitLogger("debug")
 	scheme := runtime.NewScheme()
 	corev1.AddToScheme(scheme)
 	client := fake.NewClientBuilder().WithScheme(scheme).Build()
-	var operatorClient operatorClient = operatorClient{
-		Client: client,
-	}
+
+	BeforeEach(func() {
+		operatorClient.Client = client
+	})
 
 	Context("CreateNamespace", func() {
 		When("creating a namespace", func() {
-			ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
 			It("should succeed", func() {
+				ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
 				Expect(err).ToNot(HaveOccurred())
 				Expect(ns).ToNot(BeNil())
 			})
 		})
 		When("creating a namespace that already exists", func() {
 			JustBeforeEach(func() {
+				operatorClient.CreateNamespace(context.TODO(), "testns")
+			})
+			It("should error", func() {
 				ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
-				It("should error", func() {
-					Expect(err).To(HaveOccurred())
-					Expect(err).To(MatchError("could not create namespace: testns: namespaces \"testns\" already exists"))
-					Expect(ns).To(BeNil())
-				})
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError("could not create namespace: testns: namespaces \"testns\" already exists"))
+				Expect(ns).To(BeNil())
 			})
 		})
 	})
 	Context("DeleteNamespace", func() {
 		When("deleting the existing namespace", func() {
-			err := operatorClient.DeleteNamespace(context.TODO(), "testns")
 			It("should succeed", func() {
+				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
 		When("deleting a namespace that does not exist", func() {
 			JustBeforeEach(func() {
+				operatorClient.DeleteNamespace(context.TODO(), "testns")
+			})
+			It("should return error", func() {
 				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
-				It("should return error", func() {
-					Expect(err).To(MatchError("could not delete namespace: testns: namespaces \"testns\" not found"))
-				})
+				Expect(err).To(MatchError("could not delete namespace: testns: namespaces \"testns\" not found"))
```

Comment:
I don't think we care _what_ the error is, just that it happens. Instead, I would just wrap the previous line in 

```
Expect(...).To(HaveOccurred())
```